### ISSUE TITLE
Get rid of verbose index files

### ISSUE THREAD

**Issue by mmnathan** [mediamonks/cra-template#14]
We should get rid of index files that do nothing more than re-export a module (default). Its seemingly only 'benefit' is that your import paths look nicer, but it has several downsides too:

- It’s more work to set up
- It introduces another maintenance layer
- Re-exporting types is often not possible because the build setup wont allow it
- It makes the reference to a module export more ambiguous because there’s now more than one 'origin'
- It adds many `index.ts` files to a project making 'actual' index files less discoverable

An example is the App directory with an App.tsx file, and an index.ts file which only purpose is to re-export the contents of the App.tsx file

**Comment by skulptur** [comment]
I agree with Nathan 1000%. They don't provide any real benefit and they have many real downsides.
